contrib/scripts/unloaddb.sh

```diff
@@ -0,0 +1,468 @@
+	fi
+}
+
+#
+# output: 2 string array, table_selected [], table_size []
+# Phase1: get all table names, if -i option is not given. Fill it to table_selected []
+# Phase2: for all tables ${table_selected}, calculate the size of the table, and
+# 	    fill it to table_size
+#
+function analyze_table_info ()
+{
+	local found=0
+	local line
+	local idx=0
+	local table_name
+	local db=$database
+	local query="show tables"
+
+	# Phase1: Get all table names if $from_file equals 0, otherwise from input file
+	# and fill it to array table_selected []
+
+	if [ $from_file -eq 1 ];then  # Read table name from file
+		result=$(cat $filename)
+	else
+		result=$(csql $user $pass -c "$query" $db)
+	fi
+
+	for token in $result
+	do
+		if  [ $from_file -eq 0 ] && [ $found -eq 0 ];then
+			str=$(echo $token | grep "====") # skip until we found table names in csql
+
+			if [ $? -eq 0 ];then
+				found=1
+			fi
+
+			continue
+		fi
+
+		# if table name comes from csql it is the pattern of 'code'
+		# and end with string like this
+		# "\n16 rows selected. (0.009125 sec) Committed."
+		# If it comes from file it is the pattern of code
+		#
+
+		if [ $from_file -eq 0 ] && [ ${token:0:1} != "'" ];then
+			break
+		fi
+
+		table_selected[idx]="$token"
+
+		if [ $from_file -eq 0 ];then        # remove single quota from csql output
+			table_selected[idx]=$(echo ${table_selected[idx]} | sed "s/[\']//g")
+		fi
+
+		let "idx++"
+	done
+
+	# 
+	# Phase2: calculate all table size in array table_selected[]
+	# calculate the table size to array table_size []
+	#
+
+	for ((i = 0; i < ${#table_selected[@]}; i++))
+	do
+		table_name=${table_selected[i]}
+		this_table_size=$(get_table_size $table_name)
+
+		if [ -z $this_table_size ] || [ $this_table_size -lt 0 ];then
+			echo "'$table_name': unknown table ($this_table_size)"
+			exit
+		fi
+
+		table_size[$num_tables]="$this_table_size"
+		let "num_tables++"
+	done
+}
+
+# MAIN
+#
+trap "cleanup" SIGHUP SIGINT SIGTERM
+set -o monitor
+
+get_options "$@"
+
+if [ $num_args_remain -ne 1 ] || [ -z $database ];then
+	show_usage
+	exit 1
+fi
+
+if [ $num_proc -gt $max_num_proc ];then
+	echo "Num Proc exeed Max Proc. Force set num proc to $max_num_proc"
+	num_proc=$max_num_proc
+fi
+
+if [ ! -d $target_dir ];then
+	echo "$target_dir: directory not exists or permission denied"
+	exit
+fi
+
+# make full path from "/"
+if [ ! -z $filename ] && [ ${filename:0:1} != "/" ];then
+	filename="$cwd/"$filename
+fi
+
+is_db_server_running $database
+
+logdir=$cwd"/"$database"_"unloaddb.log
+if [ -d $logdir ];then
+	rm -rf $logdir
+fi
+
+mkdir $logdir
+
+verify_user_pass
+
+echo -n "Analyzing table spacace ..."
+
+analyze_table_info
+
+if [ $num_tables -eq 0 ];then
+	echo "No Table SELECTED."
+	exit
+fi
+
+echo ""
+
+# Assign all tables to unloaddb slot
+# Assign and calculate total table size & num tables assigned in each slot
+for ((i = 0; i < $num_tables; i++))
+do
+	this_slot=$(find_least_loaded_proc)
+	tbl=${table_selected[i]}
```

Comment:
I think that tbl variable doesn't use at codes. Can you remove that ?

Reply:
That line was removed, thank you for your comment!